### PR TITLE
Valid Sudoku Leetcode solution

### PR DESCRIPTION
**Pull request by vigneshshiv**
Added Valid Sudoku solution, Description was not added through converter, it's directly added, Take a look if it's good to merge.

### CODE REVIEW

File: solutions/0000-0099/0036-valid-sudoku-easy.md [hunk 1]
```diff
@@ -1,5 +1,168 @@
...
+        int sqrt = (int) Math.sqrt(board.length);
+        int rowStart = row - row % sqrt;
+        int colStart = col - col % sqrt;
+        for (int r = rowStart; r < rowStart + sqrt; r++) {
+            for (int c = colStart; c < colStart + sqrt; c++) {
+                if (row != r && col != c && board[r][c] == num) {
+                    return false;
+                }
+            }
+        }
```
Comment (wingkwong) [inline solutions/0000-0099/0036-valid-sudoku-easy.md:161]: Need detailed explanations for beginners.

File: solutions/0000-0099/0036-valid-sudoku-easy.md [hunk 2]
```diff
@@ -1,5 +1,168 @@
...
+```
+
+</TabItem>
+</Tabs>
+
+## Approach 2: Standard
+
+Iterate over rows and columns and `3 x 3` block to check if the number is repeats. 
+
+Time complexity: O(9^n^2)
```
Comment (wingkwong) [inline solutions/0000-0099/0036-valid-sudoku-easy.md:118]: 9^n^2???

File: solutions/0000-0099/0036-valid-sudoku-easy.md [hunk 3]
```diff
@@ -1,5 +1,168 @@
...
+* `board[i].length == 9`
+* `board[i][j]` is a digit `1-9` or `'.'`.
+
+## Approach 1: HashSet
+
+Every value in the row or column or `3 x 3` block, the value is unique, With the help of this simple logic, add the Numbers in the HashSet, Add `num` value for `m`th row and for `n`th column and the same for `3 x 3` matrix. 
+
+If the same value repeat again for the same row or column or `3 x 3` block, then the given sudoku is not valid, otherwise is valid.
+
+Time Complexity: O(m * n), where m - # of rows, n - # of cols
```
Comment (wingkwong) [inline solutions/0000-0099/0036-valid-sudoku-easy.md:84]: are u using 0 for Big O notation?

**Review by wingkwong — CHANGES_REQUESTED** [review]
1. Suggest to run it locally to verify first. Don't count on the markdown previewer. Hard to read for Time Complexity & Space Complexity.

image

2. it would be nice to show in [LaTeX](https://docusaurus.io/docs/markdown-features/math-equations). e.g. O(m * n) -> $O(m * n)$.

File: solutions/0000-0099/0036-valid-sudoku-easy.md (hunk 1 above)
Comment (vigneshshiv) [reply solutions/0000-0099/0036-valid-sudoku-easy.md:161]: Hello wingwong, 
The solution posted here are accepted in Leetcode for the problem and I do maintain a repo for this. Please take a look for this problem solution on this - https://github.com/vigneshshiv/leetcode/blob/main/code/java/backtracking/ValidSudoku.java

Sure, about the Time and Space complexity, I will change it and raise a PR.

The last one about, searching the value on 3 x 3 grid, this helps to find starting position on which row and on which column. I will add the explanation..

File: solutions/0000-0099/0036-valid-sudoku-easy.md (hunk 3 above)
Comment (vigneshshiv) [reply solutions/0000-0099/0036-valid-sudoku-easy.md:84]: I'm using a Big O notation, anyway i will change with LaTex for easy to read.

File: solutions/0000-0099/0036-valid-sudoku-easy.md (hunk 2 above)
Comment (vigneshshiv) [reply solutions/0000-0099/0036-valid-sudoku-easy.md:118]: Time complexity mentioned for this problem is wrong, O(9^n^2) is actually for Sudoku Solver problem. 

Correct Time complexity for this problem is O(n^2)

**Comment by vigneshshiv** [comment]
As suggested, modified the file and added explanation. Please take a look.

**Review by wingkwong — APPROVED** [review]

**Pull request merged**